controller/src/lib.rs

```diff
@@ -569,6 +569,72 @@ impl Controller {
         }
     }
 
+    /// Enable the hot swap controller for a set of transceiver modules.
+    pub async fn enable_power(&self, modules: ModuleId) -> Result<(), Error> {
+        self.no_payload_request(modules, HostRequest::EnablePower)
+            .await?;
+        Ok(())
+    }
+
+    /// Disable the hot swap controller for a set of transceiver modules.
+    pub async fn disable_power(&self, modules: ModuleId) -> Result<(), Error> {
+        self.no_payload_request(modules, HostRequest::DisablePower)
+            .await?;
+        Ok(())
+    }
+
+    /// Assert reset for a set of transceiver modules.
+    pub async fn assert_reset(&self, modules: ModuleId) -> Result<(), Error> {
+        self.no_payload_request(modules, HostRequest::AssertReset)
+            .await?;
+        Ok(())
+    }
+
+    /// Deassert reset for a set of transceiver modules.
+    pub async fn deassert_reset(&self, modules: ModuleId) -> Result<(), Error> {
+        self.no_payload_request(modules, HostRequest::DeassertReset)
+            .await?;
+        Ok(())
+    }
+
+    /// Assert lpmode for a set of transceiver modules.
```

Comment:
I might add a note here that this is about the `LPMode` pin itself, and will have no effect if the software override of power control is set.